### DISCUSSION TITLE
Help using turtle.getItemDetail()

### DISCUSSION THREAD

**Discussion by SoundMeister** [cc-tweaked/CC-Tweaked#1565] [Q&A] [answered]
So im trying to get the turtle to identify any wheat_seed in the inventory and place it into slot 1 before dumping the rest of the inventory. What i was thinking was to use the output of turtle.getItemDetail() to identify the seeds but im not sure how to utilize the outputs yet as im still learning lua and am unsure how to recall output data to be used in a comparison

**Comment by Lupus590** [comment, marked as answer]
My best advice would be to play around in the Lua REPL[^1] prompt and get it to pretty print the result of get item detail.

I'm expecting you to do this in the Lua prompt so we are not using local, the interactive prompt doesn't handle local the same way that programs do[^2].

```lua
 pretty = require("cc.pretty")
 item = turtle.getItemDetail()
 pretty.pretty_print(item)
 ```
 
 From there you'll see a table printed out, the value that you are interested in is stored in the name key. To access that in your program you'll probably do something like this.
 
 ```lua
local  item = turtle.getItemDetail()
if item and item.name == "minecraft:wheat_seeds" then -- Note that I might have the internal name for the seeds wrong, I'm doing that from memory
  -- whatever you wanted to do with the seeds
end
```

`if item and ...`?
`turtle.getItemDetail()` returns `nil` when there is no item in that slot, this will break the indexing of `item.name` and crash the program so we make sure that item is not nil first. [^3]

[^1]: Read Execute Print Loop
[^2]: the Lua program treats each line of code as a separate function so local will make it only available to that line of code, we want to access these variables later so we have to pollute the program environment. Normally this is frowned upon in programs as it's considered bad practice but it's the correct way to use the interactive REPL prompt.
[^3]: Anything that is not `false` or `nil` is truth and while not equal to true will be resolved as such when in an if statement without a comparison operator (`==`, `>`, `<`, etc.).

**Reply by SoundMeister** [reply]
that worked perfectly, thankyou so much as i couldn't find much more detail on the function

**Reply by Lupus590** [reply]
There's quite a bit in the docs, but they are not the easiest to read for people new to programming.

https://tweaked.cc/
